server/base/src/main/java/org/apache/accumulo/server/client/ClientServiceHandler.java

```diff
@@ -194,10 +194,14 @@ public void grantSystemPermission(TInfo tinfo, TCredentials credentials, String
   }
 
   @Override
-  public void grantTablePermission(TInfo tinfo, TCredentials credentials, String user, String tableName, byte permission) throws ThriftSecurityException,
-      ThriftTableOperationException {
+  public void grantTablePermission(TInfo tinfo, TCredentials credentials, String user, String tableName, byte permission) throws TException {
     String tableId = checkTableId(instance, tableName, TableOperation.PERMISSION);
-    String namespaceId = Tables.getNamespaceId(instance, tableId);
+    String namespaceId;
+    try {
+      namespaceId = Tables.getNamespaceId(instance, tableId);
+    } catch (TableNotFoundException e) {
+      throw new TException(e);
```

Comment:
Good question, I am not sure.  I will try and test this case to see what gets printed.